tiledb/sm/serialization/rest_version.cc

```diff
@@ -0,0 +1,194 @@
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ * This file defines serialization for REST version information.
+ */
+
+#include <sstream>
+
+// clang-format off
+#ifdef TILEDB_SERIALIZATION
+#include <capnp/compat/json.h>
+#include <capnp/message.h>
+#include <capnp/serialize.h>
+#endif
+// clang-format on
+
+#include "tiledb/sm/enums/serialization_type.h"
+#include "tiledb/sm/rest/rest_client.h"
+#include "tiledb/sm/rest/rest_client_remote.h"
+#include "tiledb/sm/serialization/rest_version.h"
+#include "tiledb/sm/storage_manager/context.h"
+
+using namespace tiledb::common;
+
+namespace tiledb::sm::serialization {
+
+class RestVersionSerializationException : public StatusException {
+ public:
+  explicit RestVersionSerializationException(const std::string& message)
+      : StatusException("[TileDB::Serialization][RestVersion]", message) {
+  }
+};
+
+class RestVersionSerializationDisabledException
+    : public RestVersionSerializationException {
+ public:
+  explicit RestVersionSerializationDisabledException()
+      : RestVersionSerializationException(
+            "Cannot (de)serialize; serialization not enabled.") {
+  }
+};
+
+#ifdef TILEDB_SERIALIZATION
+
+void rest_version_to_capnp(
+    Context* ctx, capnp::RestVersion::Builder* rest_version_builder) {
+  if (ctx->has_rest_client()) {
+    // The REST version is initialized in the RestClientRemote constructor.
+    rest_version_builder->setTiledbVersion(
+        static_cast<RestClientRemote&>(ctx->rest_client()).rest_version());
+  } else {
+    // We should not hit this case, if TILEDB_SERIALIZATION is enabled there
+    // will be a rest client attached to ContextResources.
+    throw RestVersionSerializationException(
+        "Cannot serialize REST version with no initialized RESTClient.");
+  }
+}
+
+std::string rest_version_from_capnp(
+    const capnp::RestVersion::Reader& rest_version_reader) {
+  if (rest_version_reader.hasTiledbVersion()) {
+    return rest_version_reader.getTiledbVersion();
+  }
+  // TODO: What is best to return here?
+  return "2.X.0";
+}
+
+void rest_version_serialize(
```

Comment:
Since we no longer use those for testing or in the REST server, I'd suggest to remove them so that we don't add dead code.